command-line/lib/Command/Key/Public.hs

```diff
@@ -40,13 +43,19 @@ mod liftCmd = command "public" $
         <> progDesc "Get the public counterpart of a private key"
         <> footerDoc (Just $ string $ mconcat
             [ "The private key is read from stdin."
+            , "Extended public key is default. In order to get public key"
+            , ", ei., without chain code use '--pub'."
             ])
   where
     parser = Public
         <$> encodingOpt [humanReadablePart|xpub|]
+        <*> publicOpt
```

Comment:
Sometthing like this will let you change the HRP...

```suggestion
  parser = do
    pub <- publicOpt
    Public <$> encodingOpt (if pub == WithChainCode then [humanReadablePart|xpub|] else [humanReadablePart|pub|])
        <*> pure pub
```

(clean it up though)

Reply:
thanks for suggestion! in the end solved this like that https://github.com/input-output-hk/cardano-addresses/pull/87/commits/4ad1a31aa59c09476b4ae27c6465833cc28425e3

Reply:
if you use monadic rather than applicative then the ??? placeholder is not needed.